app/code/community/Bolt/Boltpay/Helper/CatalogHelper.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+class Bolt_Boltpay_Helper_CatalogHelper extends Mage_Core_Helper_Abstract
+{
+    /**
+     * @return Mage_Core_Model_Abstract
+     */
+    public function getSession()
+    {
+        return Mage::getSingleton('catalog/session');
+    }
+
+    /**
+     * Get Last real order ID from the catalog session
+     * @return integer
+     */
+    public function getLastRealOrderId()
+    {
+        return Mage::getSingleton('catalog/session')->getLastRealOrderId();
+    }
+
+    /**
+     * @param string $scope
+     * @return string
+     */
+    private function getQuoteIdKey($scope = 'ppc')
+    {
+        return $scope.'_quote_id_' . Mage::app()->getStore()->getId();
+    }
+
+    /**
+     * Get Quote for Product page
+     * @return Mage_Sales_Model_Quote
+     */
+    public  function getQuote()
+    {
+        $hasOrder = false;
+        $orderQuoteId = false;
+        $ppcQuoteId = $this->getSession()->getData($this->getQuoteIdKey());
+        if ($ppcQuoteId) {
+            $order = Mage::getModel('sales/order')->getCollection()->addAttributeToFilter('entity_id', $this->getLastRealOrderId())->setPageSize(1)->getFirstItem();
+            if ($order instanceof Mage_Sales_Model_Order) {
+                $orderQuoteId = $order->getQuoteId();
+            }
+            /** @var Mage_Sales_Model_Quote $ppcQuote */
+            $ppcQuote = Mage::getModel('sales/quote')->loadByIdWithoutStore($ppcQuoteId);
+            if ($orderQuoteId && $ppcQuoteId == $orderQuoteId) {
+                $ppcQuote->setIsActive(false);
+                $ppcQuote->delete();
+                $hasOrder = true;
+            }
+        }
+        if ($hasOrder || empty($ppcQuote)) {
```

Comment:
Please init `ppcQuote` variable before `if ($ppcQuoteId) {` to avoid warning in production servers.

Reply:
@YevhenShBolt great thanks! You are right!